mongo/integration/initial_dns_seedlist_discovery_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/connstring"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/description"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+)
+
+const (
+	seedlistDiscoveryTestsDir = "../../data/initial-dns-seedlist-discovery"
+)
+
+type seedlistTest struct {
+	URI     string   `bson:"uri"`
+	Seeds   []string `bson:"seeds"`
+	Hosts   []string `bson:"hosts"`
+	Error   bool     `bson:"error"`
+	Options bson.Raw `bson:"options"`
+}
+
+func TestInitialDNSSeedlistDiscoverySpec(t *testing.T) {
+	mtOpts := mtest.NewOptions().Topologies(mtest.ReplicaSet).CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	for _, file := range jsonFilesInDir(mt, seedlistDiscoveryTestsDir) {
+		mt.RunOpts(file, noClientOpts, func(mt *mtest.T) {
+			runSeedlistDiscoveryTest(mt, path.Join(seedlistDiscoveryTestsDir, file))
+		})
+	}
+}
+
+func runSeedlistDiscoveryTest(mt *mtest.T, file string) {
+	content, err := ioutil.ReadFile(file)
+	assert.Nil(mt, err, "ReadFile error for %v: %v", file, err)
+
+	var test seedlistTest
+	err = bson.UnmarshalExtJSONWithRegistry(specTestRegistry, content, false, &test)
+	assert.Nil(mt, err, "UnmarshalExtJSONWithRegistry error: %v", err)
+
+	if runtime.GOOS == "windows" && file == "two-txt-records" {
+		mt.Skip("skipping to avoid windows multiple TXT record lookup bug")
+	}
+	if strings.HasPrefix(runtime.Version(), "go1.11") && file == "one-txt-record-multiple-strings" {
+		mt.Skip("skipping to avoid go1.11 problem with multiple strings in one TXT record")
+	}
+
+	cs, err := connstring.Parse(test.URI)
```

Comment:
This can't really leverage `mongotest` because the tests require examining fields in the `Connstring` after it's parsed, which isn't possible if we create a `Client`.